app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Users
+  class RegistrationsController < Devise::RegistrationsController
+    # before_action :configure_sign_up_params, only: [:create]
+    # before_action :configure_account_update_params, only: [:update]
+    skip_before_action :authenticate_subdomain, only: :new
+    # GET /resource/sign_up
+
+    # POST /resource
+    def create
+      company = Company.new(company_params)
+      user = company.users.build(user_params)
+      user.role_id = User::ROLES[:Admin]
+      redirect_to new_user_session_url(subdomain: company.subdomain) if company.save!
+    end
+
+    # GET /resource/edit
+    # def edit
+    #   super
+    # end
+
+    # PUT /resource
+    # def update
+    #   super
+    # end
+
+    # DELETE /resource
+    # def destroy
+    #   super
+    # end
+
+    # GET /resource/cancel
+    # Forces the session data which is usually expired after sign
+    # in to be expired now. This is useful if the user wants to
+    # cancel oauth signing in/up in the middle of the process,
+    # removing all OAuth session data.
+    # def cancel
+    #   super
+    # end
+
+    # protected
+
+    # If you have extra params to permit, append them to the sanitizer.
+    # def configure_sign_up_params
+    #   devise_parameter_sanitizer.permit(:sign_up, keys: [:attribute])
+    # end
+
+    # If you have extra params to permit, append them to the sanitizer.
+    # def configure_account_update_params
+    #   devise_parameter_sanitizer.permit(:account_update, keys: [:attribute])
+    # end
+
+    # The path used after sign up.
+    # def after_sign_up_path_for(resource)
+    #   super(resource)
+    # end
+
+    # The path used after sign up for inactive accounts.
+    # def after_inactive_sign_up_path_for(resource)
+    #   super(resource)
+    # end
+
+    private
+
+    def company_params
+      params.require(:user).require(:company_attributes).permit(:name, :subdomain)
+    end
+
+    def user_params
+      params.require(:user).permit(:email, :password, :password_confirmation)
```

Comment:
we can add company_attributes: [:name, :subdomain] in user params to build user and company with single query in create method